private/bufpkg/bufmodule/bufmodule.go

```diff
@@ -211,10 +212,11 @@ func ModuleWithWorkspaceDirectory(workspaceDirectory string) ModuleOption {
 // from a lock file in the read bucket.
 func NewModuleForBucket(
 	ctx context.Context,
+	logger *zap.Logger,
```

Comment:
It's not OK to take a `logger` in the constructor for `Module` here - we don't take loggers in entity constructors. You'll want to figure out how to test for this somewhere else.

Reply:
Added a `buflock.CheckDeprecatedDigests`, see PR description